lib/adapters/cryptopus_adapter.rb

```diff
@@ -36,10 +36,10 @@ def save_secrets(secrets)
       secret_account = secret.to_account
       secret_account.folder = session_adapter.selected_folder.id
 
-      persisted_secret = find_account_by_name(secret.name)
-      if persisted_secret
+      begin
+        persisted_secret = find_account_by_name(secret.name)
         patch("accounts/#{persisted_secret.id}", secret_account.to_json)
-      else
+      rescue CryptopusAccountNotFoundError
```

Comment:
interesting approach for handling this update vs. create situation. 
I think it's not very common in ruby to use exceptions for this case. In other languages like for example Java it's pretty common though. 
Would love to share some thoughts with you about this pattern. 
Anyway, keep it as it is ... it's not wrong for sure !